pkg/protocol/request.go

```diff
@@ -194,7 +194,7 @@ func (req *Request) Reset() {
 	req.ResetSkipHeader()
 	req.CloseBodyStream()
 
-	req.options = nil
+	req.Options().Reset()
```

Comment:
will break the init way of req.options since it will skip the `preDefinedOpts` logic at the next round.